db/db_tunables.c

```diff
@@ -0,0 +1,1181 @@
+        DO_VERIFY(t, &num);
+
+        if (t->update) {
+            DO_UPDATE(t, &num);
+        } else {
+            *(int *)t->var = num;
+        }
+
+        logmsg(LOGMSG_DEBUG, "Tunable '%s' set to %d\n", t->name, num);
+        break;
+    }
+    case TUNABLE_DOUBLE: {
+        double num;
+        PARSE_TOKEN;
+
+        if ((ret = parse_double(buf, &num))) {
+            logmsg(LOGMSG_ERROR, "Invalid argument for '%s'.\n", t->name);
+            return 1;
+        }
+
+        /*
+          Verify the validity of the specified argument. We perform this
+          check for all DOUBLE types.
+        */
+        if ((t->flags & SIGNED) == 0) {
+            if (((t->flags & NOZERO) != 0) && (num <= 0)) {
+                logmsg(LOGMSG_ERROR,
+                       "Invalid argument for '%s' (should be > 0).\n", t->name);
+                return 1;
+            } else if (num < 0) {
+                logmsg(LOGMSG_ERROR,
+                       "Invalid argument for '%s' (should be >= 0).\n",
+                       t->name);
+                return 1;
+            }
+        }
+
+        /* Perform additional checking if defined. */
+        DO_VERIFY(t, &num);
+
+        if (t->update) {
+            DO_UPDATE(t, &num);
+        } else {
+            *(double *)t->var = num;
+        }
+
+        logmsg(LOGMSG_DEBUG, "Tunable '%s' set to %f\n", t->name, num);
+        break;
+    }
+    case TUNABLE_BOOLEAN: {
+        int num;
+        PARSE_TOKEN;
+
+        if ((ret = parse_bool(buf, &num))) {
+            logmsg(LOGMSG_ERROR, "Invalid argument for '%s'.\n", t->name);
+            return 1;
+        }
+
+        /* Perform checking if defined. */
+        DO_VERIFY(t, &num);
+
+        if (t->update) {
+            DO_UPDATE(t, &num);
+        } else {
+            *(int *)t->var = num;
+        }
+
+        logmsg(LOGMSG_DEBUG, "Tunable '%s' set to %s\n", t->name,
+               (num) ? "ON" : "OFF");
+        break;
+    }
+    case TUNABLE_STRING: {
+        PARSE_TOKEN;
+        DO_VERIFY(t, buf);
+
+        if (t->update) {
+            DO_UPDATE(t, buf);
+        } else {
+            free(*(char **)t->var);
+            *((char **)t->var) = strdup(buf);
+        }
+
+        logmsg(LOGMSG_DEBUG, "Tunable '%s' set to %s\n", t->name, value);
+        break;
+    }
+    /* ENUM types must have at least value and update functions defined. */
+    case TUNABLE_ENUM: {
+        PARSE_TOKEN;
+        DO_VERIFY(t, buf);
+        assert(t->update);
+        DO_UPDATE(t, buf);
+        break;
+    }
+    default: assert(0);
+    }
+
+    /* Check/warn for unparsed/unexpected junk in the value. */
+    tok = segtok((char *)value, value_len, &st, &ltok);
+    if (ltok != 0) {
+        logmsg(LOGMSG_WARN,
+               "Found junk in the value supplied for tunable '%s'\n", t->name);
+    }
+    return 0;
+}
+
+/*
+  Update the tunable at runtime.
+
+  @return
+    0           Success
+    1           Failure
```

Comment:
Good point. Though this change is not needed now, but it will be once I fix process_message to use handle_runtime_tunable(). Fixed.